### PR TITLE
Add Recent Activities in Home Page

### PR DESCRIPTION
**Pull request by subinasr**

### CODE REVIEW

**Review by tnagorra — APPROVED** [review]

**Review by tnagorra — APPROVED** [review]

File: src/views/Home/RecentActivity/index.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    const [
+        ,
+        recentActivitiesResponse,
+        ,
+        ,
+    ] = useRequest<MultiResponse<RecentActivity>>({
+        url: 'server://projects/recent-activities/',
+        method: 'GET',
+        autoTrigger: true,
+        onSuccess: response => console.warn('Recent Activities fetch successfully', response),
```
Comment (AdityaKhatri) [inline src/views/Home/RecentActivity/index.tsx]: Remove console warn

File: src/views/Home/RecentActivity/index.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    projectDisplayName: string;
+    createdBy: number;
+    type: string;
+    createdByDisplayName: string;
+}
+
+const keySelector = (d: RecentActivity) => d.id;
+const emptyLink = '#';
+
+function ActivityRenderer(props: RecentActivity) {
```
Comment (AdityaKhatri) [inline src/views/Home/RecentActivity/index.tsx]: Separate into a new page

File: src/views/Home/RecentActivity/styles.scss [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    flex-direction: column;
+
+    .header {
+        padding: var(--dui-spacing-small) 0;
+    }
+
+    .content-container {
+        display: flex;
+        flex-direction: column;
+        border-radius: 10px;
```
Comment (AdityaKhatri) [inline src/views/Home/RecentActivity/styles.scss]: If Card component is used this is unnecessary.

File: src/views/Home/RecentActivity/index.tsx [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    createdAt: string;
+    project: number;
+    projectDisplayName: string;
+    createdBy: number;
+    type: string;
+    createdByDisplayName: string;
+}
+
+const keySelector = (d: RecentActivity) => d.id;
+const emptyLink = '#';
```
Comment (AdityaKhatri) [inline src/views/Home/RecentActivity/index.tsx]: Add a TODO to change this link after the they are created.

File: src/views/Home/RecentActivity/index.tsx [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        createdByDisplayName: info.createdByDisplayName,
+        type: info.type,
+    }), []);
+
+
+    return (
+        <Card className={styles.recentActivity}>
+            <Header
+                className={styles.header}
+                heading="Recent Activities"
```
Comment (AdityaKhatri) [inline src/views/Home/RecentActivity/index.tsx]: Use translation here

File: src/views/Home/RecentActivity/index.tsx [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+function RecentActivity() {
+    const [
+        ,
+        recentActivitiesResponse,
+        ,
+        ,
+    ] = useRequest<MultiResponse<RecentActivity>>({
+        url: 'server://projects/recent-activities/',
+        method: 'GET',
+        autoTrigger: true,
```
Comment (AdityaKhatri) [inline src/views/Home/RecentActivity/index.tsx:31]: Add polling behavior here to continuously fetch new data.

File: src/views/Home/RecentActivity/index.tsx (hunk 1 above)
Comment (AdityaKhatri) [inline src/views/Home/RecentActivity/index.tsx]: handle request failure

File: src/views/Home/RecentActivity/index.tsx (hunk 5 above)
Comment (AdityaKhatri) [inline src/views/Home/RecentActivity/index.tsx]: show small loading animation on the side of header when new data is being pulled

File: src/views/Home/RecentActivity/index.tsx [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+        projectDisplayName,
+        createdAt,
+        type,
+    } = props;
+
+    return (
+        <div className={styles.activityItem}>
+            <div className={styles.displayPicture}>
+                <DisplayPicture />
+            </div>
```
Comment (AdityaKhatri) [inline src/views/Home/RecentActivity/index.tsx]:
```suggestion
            <DisplayPicture className={styles.displayPicture} />
```

File: src/views/Home/RecentActivity/styles.scss [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+            .activity-item {
+                display: flex;
+                align-items: center;
+                border-bottom: var(--width-separator-thin) solid var(--dui-border-color-card);
+                padding: calc(var(--dui-spacing-extra-large) - var(--dui-spacing-small));
+
+                .display-picture {
+                    margin-right: var(--spacing-medium);
+                    border-radius: 50%;
```
Comment (AdityaKhatri) [inline src/views/Home/RecentActivity/styles.scss]: Isn't this done internally by DisplayPicture

File: src/views/Home/RecentActivity/index.tsx [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+                            ? _ts('recentActivity', 'leadAdded')
+                            : _ts('recentActivity', 'entryCommentAdded')
+                        }
+                    </span>
+                    <a
+                        className={styles.link}
+                        href={emptyLink}
+                    >
+                        {projectDisplayName}
+                    </a>
```
Comment (AdityaKhatri) [inline src/views/Home/RecentActivity/index.tsx]: Use Link component from dui

File: src/views/Home/RecentActivity/index.tsx [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+                </span>
+                <div className={styles.dateContainer}>
+                    <FormattedDate
+                        className={styles.date}
+                        value={createdAt}
+                        mode="hh:mm aaa, MMM dd, yyyy"
+                    />
+                </div>
```
Comment (AdityaKhatri) [inline src/views/Home/RecentActivity/index.tsx]: Try to use formatted date directly without an extra container

File: src/views/Home/RecentActivity/index.tsx [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+        ,
+    ] = useRequest<MultiResponse<RecentActivity>>({
+        url: 'server://projects/recent-activities/',
+        method: 'GET',
+        autoTrigger: true,
+        onSuccess: response => console.warn('Recent Activities fetch successfully', response),
+    });
+
+    const activityRendererParams = useCallback((_: number, info: RecentActivity) => ({
+        id: info.id,
```
Comment (AdityaKhatri) [inline src/views/Home/RecentActivity/index.tsx]: it is better to name it `activityId` to make it a bit more clear.

File: src/views/Home/RecentActivity/styles.scss (hunk 3 above)
Comment (subinasr) [reply src/views/Home/RecentActivity/styles.scss]: The Card component does not have any styling related to border-radius. When not used, the box is just a rectangle with pointed corners.

File: src/views/Home/RecentActivity/styles.scss (hunk 8 above)
Comment (subinasr) [reply src/views/Home/RecentActivity/styles.scss]: This is what it looks if we don't use a wrapper
image

**Pull request merged**